redash/handlers/users.py

```diff
@@ -217,6 +217,19 @@ def post(self, user_id):
 
         return user.to_dict(with_api_key=is_admin_or_owner(user_id))
 
+    @require_admin
+    def delete(self, user_id):
+        user = models.User.get_by_id_and_org(user_id, self.current_org)
```

Comment:
Sounds like this could suffice (without a try/except), no?

```
if !user.is_invitation_pending:
    abort(403, message="You cannot delete activated users. "
                       "Please disable the user instead.")

Reply:
Yes.